src/Firebase/Exception/AuthApiExceptionConverter.php

```diff
@@ -108,6 +111,18 @@ private function convertGuzzleRequestException(RequestException $e): AuthExcepti
             return new PhoneNumberExists('The phone number is already in use by another account.', $code, $e);
         }
 
+        if (\mb_stripos($message, 'invalid_idp_response') !== false) {
```

Comment:
I have kept the other dedicated exceptions for backward compatibility, but for the next major release, I am planning to reduce the amount a lot. In the meantime, could you please merge the three custom exception types to one single exception? I'm not sure about the best name, but at the moment I'd go with `FederatedUserLinkingFailed` or `FailedToLinkFederatedUser`. If you have additional suggestions, please let me know ^^

Reply:
I created a `LinkProviderException` because as I understand the provider is not neccesseraly federated, but maybe Im wrong. Not sure you will like the solution, but this exception handling is weird enought already so I've chosen this way.

PHPStan didn't allow me to write the assignment into the if (maybe a bug for phpstan).

Reply:
Ah, I didn't think of the "not necessarily federated"... I thought a little about it, and would like `ProviderLinkFailed` as the name of the exception (this matches the `linkProvider*` methods in the `Auth` class). You might have noticed that most exceptions don't have the `Exception` suffix in them 😓

Please also keep the message matching in the `AuthApiExceptionConverter`, please 🙏 All other exceptions are converted in there, and it removes the need for another condition/check if the static constructor of the new exception actually returned itself.